### ISSUE TITLE
Logger mode with minimal output

### ISSUE THREAD

**Issue by cvanaret** [cvanaret/Uno#42]
Uno tends to be talkative, even in the ERROR logger mode. It needs a silent logger mode with minimal output.
Reported by @odow

**Comment by odow** [comment]
It needs a silent logger mode with <s>minimal</s> _no_ output.

**Comment by cvanaret** [comment]
Would you prefer something like:
```
Problem /media/data/AMPL/CUTEst/hs015
2 variables, 2 constraints
Status:					Converged with feasible KKT point
```
or literally no output?

**Comment by odow** [comment]
Literally no output.

A fairly typical use-case of Ipopt is solving a bunch of problems in a loop with different parameter values. Another is embedding it as a sub-solver in a meta-solver like https://github.com/lanl-ansi/Juniper.jl. You don't want logging from Ipopt polluting Juniper's logs.

I guarantee you that without this, the first post on JuMP's community forum about Uno will be the question "how do I turn off printing for Uno."

**Comment by odow** [comment]
We have programatic ways of checking problem size and termination statues etc. You don't need to manually read the log unless something has gone wrong.

**Comment by cvanaret** [comment]
> I guarantee you that without this, the first post on JuMP's community forum about Uno will be the question "how do I turn off printing for Uno."

Ahahah got it. I'm on it.

**Comment by cvanaret** [comment]
Implemented in commit 4b8bcb9.
The logger levels are now: SILENT, DISCRETE, WARNING, INFO, DEBUG, DEBUG2, DEBUG3
- SILENT produces no output (the only thing I could not avoid is the line break in ASL's `write_sol`).
- DISCRETE only prints the used options and the result.
- the other levels are unchanged.

**Comment by odow** [comment]
> the only thing I could not avoid is the line break in ASL's write_sol

Not sure I follow? `write_sol` prints a `\n` to stdout? That's annoying.

**Comment by cvanaret** [comment]
Yes. ~~I think it's [this one](https://github.com/ampl/asl/blob/2f5d9de248c53a3063bba23af2013cd3db768bf8/src/solvers/writesol.c#L498). Perhaps I can redirect stdout, but this is indeed annoying.~~
It's [this one](https://github.com/ampl/asl/blob/2f5d9de248c53a3063bba23af2013cd3db768bf8/src/solvers/writesol.c#L508). I'll try to fiddle with the options to see if I can discard it.

**Comment by cvanaret** [comment]
Fixed in cf3ec48.
